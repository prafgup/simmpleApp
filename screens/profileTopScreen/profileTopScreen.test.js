import React from "react";
import {shallow} from "enzyme";



import {
    NameBio,
    renderStatusIcon,
    renderFileUri,
    renderStatusColor,
    sketchAvatar,
    svgBuilder
} from "./component";



describe("Profile Top Page",()=>{

    describe("Name and Bio",()=>{
        let component ;

        beforeEach(()=>{
            component = shallow(NameBio());
        });

        it("Should match to snapshot",()=>{
            expect(component).toMatchSnapshot("Name Bio");
        });

        it("Should Render all Details",()=>{
            expect(component.find("[data-test='profileName']").length).toBe(1);
            expect(component.find("[data-test='profession']").length).toBe(1);
            expect(component.find("[data-test='website']").length).toBe(1);
        });
    })

    describe("Decagon Points ",()=>{
        it("Should return 10 points in string",()=>{
            const points = 10;
            const height =100;
            const component = sketchAvatar(height,points);
            expect(component).toMatchSnapshot("All Points");
            expect(component.split(",")).toHaveLength(points+1);
        });
    })

    describe("Should return correct Status Color",()=>{

        it("Should return white",()=>{
            const status = null;
            const component = renderStatusColor(status);
            expect(component).toMatchSnapshot("Status Null");
            expect(component).toBe("white");
        });
        it("Should return orange",()=>{
            const status = {
                statusSeen : false
            };
            const component = renderStatusColor(status);
            expect(component).toMatchSnapshot("Status Seen False");
            expect(component).toBe("orange");
        });
        it("Should return grey",()=>{
            const status = {
                statusSeen : true
            };
            const component = renderStatusColor(status);
            expect(component).toMatchSnapshot("Status Seen True");
            expect(component).toBe("grey");
        });
    })



    describe("Should Return Correct Floating Icon",()=>{
        it("Should Return Plus Sign",()=>{
            const status = {};
            const func = jest.fn();
            const component = shallow(renderStatusIcon(status,func));
            expect(component).toMatchSnapshot("Plus Sign");
            expect(component.find("[data-test='plus']").length).toBe(1);

            component.find("[data-test='plus']").first().props().onPress();
            expect(func).toHaveBeenCalled();

        });
        it("Should Return Three Dots",()=>{
            const status = {
                statusSeen: false
            };
            const func = jest.fn();
            const component = shallow(renderStatusIcon(status,func));
            expect(component).toMatchSnapshot("Three Dots");
            expect(component.find("[data-test='threeDot']").length).toBe(1);

            

        });
        it("Should Return Just View",()=>{
            const status = {
                statusSeen: true
            };
            const func = jest.fn();
            const component = shallow(renderStatusIcon(status,func));
            expect(component).toMatchSnapshot("View");
            
        });
    });

    describe("Image Render",()=>{
        it("Should Return Default",()=>{
            const uri = null;
            const component = shallow(renderFileUri(uri));
            expect(component).toMatchSnapshot("Default Image");
           

        });
        it("Should Return uriImage",()=>{
            const uri = "testUrl";
            const component = shallow(renderFileUri(uri));
            expect(component).toMatchSnapshot("User Image");

        });
    });

    describe("SVG Image Render",()=>{
        it("Should Match Snapshot",()=>{
            const component = shallow(svgBuilder(null,null));
            expect(component).toMatchSnapshot("Default Image");

        });
    });

});