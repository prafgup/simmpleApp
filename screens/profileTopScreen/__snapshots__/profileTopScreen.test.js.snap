// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Top Page Decagon Points  Should return 10 points in string: All Points 1`] = `"100,50 90.45084971874738,79.38926261462366 65.45084971874738,97.55282581475768 34.549150281252636,97.55282581475768 9.549150281252633,79.38926261462368 0,50.000000000000014 9.549150281252627,20.61073738537635 34.54915028125262,2.447174185242323 65.45084971874736,2.447174185242318 90.45084971874738,20.610737385376332 "`;

exports[`Profile Top Page Image Render Should Return Default: Default Image 1`] = `ShallowWrapper {}`;

exports[`Profile Top Page Image Render Should Return uriImage: User Image 1`] = `ShallowWrapper {}`;

exports[`Profile Top Page Name and Bio Should match to snapshot: Name Bio 1`] = `ShallowWrapper {}`;

exports[`Profile Top Page SVG Image Render Should Match Snapshot: Default Image 1`] = `ShallowWrapper {}`;

exports[`Profile Top Page Should Return Correct Floating Icon Should Return Just View: View 1`] = `ShallowWrapper {}`;

exports[`Profile Top Page Should Return Correct Floating Icon Should Return Plus Sign: Plus Sign 1`] = `ShallowWrapper {}`;

exports[`Profile Top Page Should Return Correct Floating Icon Should Return Three Dots: Three Dots 1`] = `ShallowWrapper {}`;

exports[`Profile Top Page Should return correct Status Color Should return grey: Status Seen True 1`] = `"grey"`;

exports[`Profile Top Page Should return correct Status Color Should return orange: Status Seen False 1`] = `"orange"`;

exports[`Profile Top Page Should return correct Status Color Should return white: Status Null 1`] = `"white"`;
